src/kitovu/sync/plugin/moodle.py

```diff
@@ -51,11 +51,18 @@ def _request(self, func: str, **kwargs: str) -> typing.Any:
         assert req.status_code == 200, req  # FIXME
         data: JsonType = req.json()
         logger.debug(f'Got data: {data}')
-
-        assert 'exception' not in data, data
-        assert 'error' not in data, data
+        self._check_json_answer(data)
         return data
 
+    def _check_json_answer(self, data: JsonType) -> None:
+        if "errorcode" in data and data["errorcode"] == "invalidtoken":
```

Comment:
You can use `data.get("errorcode", None)` (`None` as fallback) here - however, since you need it twice, I'd make it a variable before the `if` anyways.